packages/teams-js/src/internal/communication.ts

```diff
@@ -46,16 +53,13 @@ class CommunicationPrivate {
   public static childMessageQueue: MessageRequest[] = [];
   public static topMessageQueue: MessageRequest[] = [];
   public static nextMessageId = 0;
-  public static callbacks: {
-    [id: number]: Function; // (arg1, arg2, ...etc) => void
-  } = {};
-  public static promiseCallbacks: {
-    [id: number]: Function; // (args[]) => void
-  } = {};
-  public static portCallbacks: {
-    [id: number]: (port?: MessagePort, args?: unknown[]) => void;
-  } = {};
+  public static callbacks: Map<MessageUUID, Function> = new Map();
+  public static promiseCallbacks: Map<MessageUUID, Function> = new Map();
```

Comment:
>Function [](http://example.com/codeflow?start=51&length=8)

Since `Function` is general and generally disallowed, could this be updated to use a type more specific to the type of function being expected? Maybe:

```
type PromiseCallback<T> = (value: T) => void;

const promiseCallbacks: Map<number, PromiseCallback<unknown>> = new Map();
```


With this solution we can lean on the type system to help ensure no one accidentally puts any other type of function into the `Map`